pkg/stream/stream_run.go

```diff
@@ -221,11 +221,16 @@ func Run(ctx context.Context, logger loglib.Logger, config *Config, instrumentat
 
 	if config.Processor.Transformer != nil {
 		logger.Info("adding transformation layer to processor...")
-		transformer, err := transformer.New(config.Processor.Transformer, processor, transformer.WithLogger(logger))
+		validationFunc := transformer.GetTransformerValidationFunc(ctx, config.Listener.Postgres.Replication.PostgresURL, config.Listener.Snapshot.Generator.URL)
```

Comment:
As mentioned in the other comments, it would be cleaner to have the checks in the `GetTransformerValidationFunc` here, by only adding the option if possible. It's also necessary to check for nil values on the config since only one listener is provided at a time.
```
opts := []transformer.Option{transformer.WithLogger(logger)}
pgURL := ""
switch {
  case config.Listener.Postgres != nil:
      pgURL = config.Listener.Postgres.Replication.PostgresURL
  case config.Listener.Snapshot != nil:
      pgURL =config.Listener.Snapshot.Generator.URL
}
if pgURL != "" {
  opts = append(opts, transformer.WithValidator(transformer.PostgresValidator(ctx, pgURL))
} 
```